src/context/chartDataContext.tsx

```diff
@@ -1,10 +1,22 @@
-import { createContext, useContext } from 'react';
+import { createContext, useContext, useEffect } from 'react';
+import { ChartData, setChartData } from '../state/chartDataSlice';
+import { useAppDispatch, useAppSelector } from '../state/hooks';
+import { RechartsRootState } from '../state/store';
 
-const ChartDataContext = createContext<any[] | undefined>(undefined);
 const DataStartIndexContext = createContext<number>(0);
 const DataEndIndexContext = createContext<number>(0);
 
-export const ChartDataContextProvider = ChartDataContext.Provider;
+export const ChartDataContextProvider = (props: { chartData: ChartData }): null => {
```

Comment:
this isn't context anymore though?

Reply:
Correct, I will migrate the other properties too and update names. Later.